### YamlMime:UniversalReference
ms.openlocfilehash: 244d32f81364753d62265cdc791e4d8a489e0883
ms.sourcegitcommit: 016e17422cf62dc9f1de1809666e9244e41cc1a9
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 11/09/2018
ms.locfileid: "26228212"
items:
- uid: office.OfficeExtension.Error
  summary: <span data-ttu-id="4e815-101">L’objet error renvoyé par `context.sync()` <!-- -->, si une promesse est refusée en raison d’une erreur lors du traitement de la demande.</span><span class="sxs-lookup"><span data-stu-id="4e815-101">The error object returned by `context.sync()`<!-- -->, if a promise is rejected due to an error while processing the request.</span></span>
  name: OfficeExtension.Error
  fullName: OfficeExtension.Error
  langs:
  - typeScript
  type: class
  package: office
  remarks: "\n#### <a name=\"examples\"></a><span data-ttu-id=\"4e815-102\">Exemples</span><span class=\"sxs-lookup\"><span data-stu-id=\"4e815-102\">Examples</span></span>\n\n```javascript\n// Run a batch operation against the Word object model.\nWord.run(function (context) {\n\n    // Create a proxy object for the document body.\n    var body = context.document.body;\n\n    // Queue a commmand to insert text in to the beginning of the body.\n    // This will cause an OfficeExtension.Error.\n    body.insertText(0);\n\n    // Synchronize the document state by executing the queued-up commands,\n    // and return a promise to indicate task completion.\n    return context.sync();\n})\n.catch(function (error) {\n    if (error instanceof OfficeExtension.Error) {\n        console.log('Error code and message: ' + error.toString());\n    }\n});\n```"
  children:
  - office.OfficeExtension.Error.code
  - office.OfficeExtension.Error.debugInfo
  - office.OfficeExtension.Error.innerError
  - office.OfficeExtension.Error.message
  - office.OfficeExtension.Error.name
  - office.OfficeExtension.Error.stack
  - office.OfficeExtension.Error.traceMessages
- uid: office.OfficeExtension.Error.code
  summary: <span data-ttu-id="4e815-103">Chaîne code d’erreur, tel que « InvalidArgument ».</span><span class="sxs-lookup"><span data-stu-id="4e815-103">Error code string, such as "InvalidArgument".</span></span>
  name: code
  fullName: code
  langs:
  - typeScript
  type: property
  syntax:
    content: 'code: string;'
    return:
      type:
      - string
- uid: office.OfficeExtension.Error.debugInfo
  summary: <span data-ttu-id="4e815-104">Les informations de débogage (utile pour détaillée de journalisation de l’erreur, c'est-à-dire via `JSON.stringify(...)` <!-- -->).</span><span class="sxs-lookup"><span data-stu-id="4e815-104">Debug info (useful for detailed logging of the error, i.e., via `JSON.stringify(...)`<!-- -->).</span></span>
  name: debugInfo
  fullName: debugInfo
  langs:
  - typeScript
  type: property
  syntax:
    content: 'debugInfo: DebugInfo;'
    return:
      type:
      - office.OfficeExtension.DebugInfo
- uid: office.OfficeExtension.Error.innerError
  summary: <span data-ttu-id="4e815-105">Erreur interne, le cas échéant.</span><span class="sxs-lookup"><span data-stu-id="4e815-105">Inner error, if applicable.</span></span>
  name: innerError
  fullName: innerError
  langs:
  - typeScript
  type: property
  syntax:
    content: 'innerError: Error;'
    return:
      type:
      - Error
- uid: office.OfficeExtension.Error.message
  summary: <span data-ttu-id="4e815-106">Le message d’erreur transmis par le biais de l’application Office hôte.</span><span class="sxs-lookup"><span data-stu-id="4e815-106">The error message passed through from the host Office application.</span></span>
  name: message
  fullName: message
  langs:
  - typeScript
  type: property
  syntax:
    content: 'message: string;'
    return:
      type:
      - string
- uid: office.OfficeExtension.Error.name
  summary: '<span data-ttu-id="4e815-107">Nom de l’erreur : « OfficeExtension.Error ».</span><span class="sxs-lookup"><span data-stu-id="4e815-107">Error name: "OfficeExtension.Error".</span></span>'
  name: name
  fullName: name
  langs:
  - typeScript
  type: property
  syntax:
    content: 'name: string;'
    return:
      type:
      - string
- uid: office.OfficeExtension.Error.stack
  summary: <span data-ttu-id="4e815-108">Pile de suivi, le cas échéant.</span><span class="sxs-lookup"><span data-stu-id="4e815-108">Stack trace, if applicable.</span></span>
  name: stack
  fullName: stack
  langs:
  - typeScript
  type: property
  syntax:
    content: 'stack: string;'
    return:
      type:
      - string
- uid: office.OfficeExtension.Error.traceMessages
  summary: <span data-ttu-id="4e815-109">Suivi des messages (le cas échéant) qui ont été ajoutés un `context.trace()` invocation avant d’appeler `context.sync()` <!-- -->.</span><span class="sxs-lookup"><span data-stu-id="4e815-109">Trace messages (if any) that were added via a `context.trace()` invocation before calling `context.sync()`<!-- -->.</span></span> <span data-ttu-id="4e815-110">Si une erreur s’est produite, il contient tous les messages de suivi qui ont été exécutées avant l’erreur s’est produite.</span><span class="sxs-lookup"><span data-stu-id="4e815-110">If there was an error, this contains all trace messages that were executed before the error occurred.</span></span> <span data-ttu-id="4e815-111">Les messages ci-dessous peuvent vous aider à contrôler la séquence de l’exécution du programme et de détecter le cas de l’erreur.</span><span class="sxs-lookup"><span data-stu-id="4e815-111">These messages can help you monitor the program execution sequence and detect the case of the error.</span></span>
  name: traceMessages
  fullName: traceMessages
  langs:
  - typeScript
  type: property
  syntax:
    content: 'traceMessages: string[];'
    return:
      type:
      - string[]
      description: "\n#### <a name=\"examples\"></a><span data-ttu-id=\"4e815-112\">範例</span><span class=\"sxs-lookup\"><span data-stu-id=\"4e815-112\">Examples</span></span>\n\n```javascript\n// The following example shows how you can instrument a batch of commands\n// to determine where an error occurred. The first batch successfully\n// inserts the first two paragraphs into the document and cause no errors.\n// The second batch successfully inserts the third and fourth paragraphs\n// but fails in the call to insert the fifth paragraph. All other commands\n// after the failed command in the batch are not executed, including the\n// command that adds the fifth trace message. In this case, the error\n// occurred after the fourth paragraph was inserted, and before adding the\n// fifth trace message.\n\n// Run a batch operation against the Word object model.\nWord.run(function (context) {\n\n    // Create a proxy object for the document body.\n    var body = context.document.body;\n\n    // Queue a commmand to insert the paragraph at the end of the document body.\n    // Start a batch of commands.\n    body.insertParagraph('1st paragraph', Word.InsertLocation.end);\n    // Queue a command for instrumenting this part of the batch.\n    context.trace('1st paragraph successful');\n\n    body.insertParagraph('2nd paragraph', Word.InsertLocation.end);\n    context.trace('2nd paragraph successful');\n\n    // Synchronize the document state by executing the queued-up commands,\n    // and return a promise to indicate task completion.\n    return context.sync().then(function () {\n        // Queue a commmand to insert the paragraph at the end of the document body.\n        // Start a new batch of commands.\n        body.insertParagraph('3rd paragraph', Word.InsertLocation.end);\n        context.trace('3rd paragraph successful');\n\n        body.insertParagraph('4th paragraph', Word.InsertLocation.end);\n        context.trace('4th paragraph successful');\n\n        // This command will cause an error. The trace messages in the queue up to\n        // this point will be available via Error.traceMessages.\n        body.insertParagraph(0, '5th paragraph', Word.InsertLocation.end);\n        // Queue a command for instrumenting this part of the batch.\n        // This trace message will not be set on Error.traceMessages.\n        context.trace('5th paragraph successful');\n    }).then(context.sync);\n})\n.catch(function (error) {\n    if (error instanceof OfficeExtension.Error) {\n        console.log('Trace messages: ' + error.traceMessages);\n    }\n});\n\n// Output: \"Trace messages: 3rd paragraph successful,4th paragraph successful\"\n```"
